### PR TITLE
fix(rpc): update StateWaitMsg against v1 spec

### PR DESCRIPTION
**Pull request by hanabi1224**
## Summary of changes

(Part of https://github.com/ChainSafe/forest/issues/4424)
This PR fixes the `StateWaitMsg` RPC method against the [v1 spec](https://github.com/filecoin-project/lotus/blob/master/documentation/en/api-v1-unstable-methods.md#StateWaitMsg) and uses it for both v0 and v1 endpoints until https://github.com/ChainSafe/forest/issues/3960 is resolved. (Keeping the v0 implementation with a different name for now)

Changes introduced in this pull request:

- update `StateWaitMsg` against v1 spec

## Reference issue to close (if applicable)

Closes

## Change checklist

- [x] I have performed a self-review of my own code,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works (if possible),
- [x] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/tool/subcommands/api_cmd.rs [hunk 1]
```diff
@@ -980,7 +980,12 @@ fn state_tests_with_tipset<DB: Blockstore>(
             tests.extend([
                 RpcTest::identity(StateReplay::request((tipset.key().into(), msg_cid))?),
                 validate_message_lookup(
-                    StateWaitMsg::request((msg_cid, 0))?.with_timeout(Duration::from_secs(30)),
```
Comment (LesnyRumcajs) [inline src/tool/subcommands/api_cmd.rs:983]: Can we test v0 as well, or is it not feasible?

Comment (hanabi1224) [reply src/tool/subcommands/api_cmd.rs:983]: Not easily feasible, the actual V0 method name uses a `V0` suffix and does not exist in lotus

Comment (LesnyRumcajs) [reply src/tool/subcommands/api_cmd.rs:983]: that's disappointing. It effectively makes the entire v0 path for this method untested and likely incorrect (now or in the future)

**Review by LesnyRumcajs — APPROVED** [review]

File: src/tool/subcommands/api_cmd.rs (hunk 1 above)
Comment (lemmih) [reply src/tool/subcommands/api_cmd.rs:983]: Right now, we can either test the v0 methods or the v1 methods but not both. Aatif is working on fixing this.

**Review by lemmih — APPROVED** [review]

**Pull request merged**